File: RECURSION/permutation.py
def solve(nums, index, ans):
    if index >= len(nums):
        ans.append(nums[:])
        return
    
    for j in range(index, len(nums)):
        nums[index], nums[j] = nums[j], nums[index]
        solve(nums, index + 1, ans)
        # backtrack
        nums[index], nums[j] = nums[j], nums[index]

nums = [2, 3, 4, 5, 6,7,8]
ans = []
index = 0

solve(nums, index, ans)

# Print the generated permutations in one line
print(*ans)

File: RECURSION/permutation.cpp
#include <bits/stdc++.h>
using namespace std;

void solve(vector<int> &nums, int index, vector<vector<int>> &ans)
{
    if (index >= nums.size())
    {
        ans.push_back(nums);
        return;
    }

    for (int j = index; j < nums.size(); j++)
    {
        swap(nums[j], nums[index]);
        solve(nums, index + 1, ans);
        // backtrack
        swap(nums[j], nums[index]);
    }
}

int main()
{
    vector<int> nums = {2, 3, 4, 5, 6, 7};
    vector<vector<int>> ans;
    int index = 0;

    solve(nums, index, ans);

    // Print the generated permutations
    for (const auto &permutation : ans)
    {
        for (int num : permutation)
        {
            cout << num << " ";
        }
    }

    return 0;
}
